src/groups/mwc/mwcst/mwcst_statcontext.h

```diff
@@ -799,6 +799,10 @@ class StatContext {
     /// all sibling subcontexts.
     int uniqueId() const;
 
+    /// Return the number of times this 'StatContext' had 'snapshot' called on
+    /// it.
+    bsls::Types::Int64 numSnapshots() const;
```

Comment:
This is useful, but also it's an interface change in `MWC`. Do we want to keep it?

Reply:
I need to [check it](https://github.com/bloomberg/blazingmq/pull/384/files#diff-03491c2d54484fc4785eb1b838445ae48b73f16c1b0b491914c21c3e5b1322dfR603-R608) in `portsDeleter()` as before the first snapshot `numSubcontexts()` returns zero, and I don't want to delete new `StatContext`s in this case. It happens because `StatContext::addSubcontext()` puts the newly created contexts to `d_newSubcontexts` vector.
https://github.com/bloomberg/blazingmq/blob/7d4b3d7caa075e5cca89995008e6fad30082b3f2/src/groups/mwc/mwcst/mwcst_statcontext.cpp#L631
Then it moves them to `d_statContext` vector on next `StatContext::snapshot()`
https://github.com/bloomberg/blazingmq/blob/7d4b3d7caa075e5cca89995008e6fad30082b3f2/src/groups/mwc/mwcst/mwcst_statcontext.cpp#L339
However `StatContext::numSubcontexts()` doesn't take into account `d_newSubcontext` and will return 0 if there haven't been any stapshots yet.
https://github.com/bloomberg/blazingmq/blob/7d4b3d7caa075e5cca89995008e6fad30082b3f2/src/groups/mwc/mwcst/mwcst_statcontext.h#L1136-L1140
So if we don't check the number of snapshots in `portsDeleter()` we can unintentionally delete the `StatContext` that has just been added.